### ISSUE TITLE
Strange behavior of nodes and links when creating Events

### ISSUE THREAD

**Issue by grotskat** [kbss-cvut/fta-fmea-ui#527]
I tested FTA for **amc1** and for **L39NG.A1**, there are 2 issues.

- [ ] **1) Issue in the FTA for amc1:**
If we have a larger fault tree, then the **BASIC** events will start skipping/moving to the top left corner with the next newly created event.
![image](https://github.com/user-attachments/assets/6a86a029-afff-42fc-bcef-b74e511711ad)

- [x] **2) Issue in the FTA for L39NG.A1:**
Links are behaving strangely.
![image](https://github.com/user-attachments/assets/6102ff9b-d071-4aca-95d5-dfd4fbc13d0f)
![image](https://github.com/user-attachments/assets/52919da9-bbf0-4009-8189-f470a753975b)

**Comment by kostobog** [comment]
@LaChope 
Regarding issue 1)
The final layout of the fault tree is a combination of:
- persistent layout information (rectangle object with properties x, y, width and height sent by server with each fault event that has persistent layout. 
- method `layout` in EditorCanvas - this method should layout nodes which have no persistent layout (as well as nodes with type `fta.ConditioningEvent`). However, the layout does not work properly when it has to layout nodes for which their parent node has persistent layout and is excluded from the layout. See, fault tree `test` for aircraft `L-39NG.A1` by `psenicka.record.adm`.

Persistent layout is applied to nodes in `renderTree` function in `RenderTree.tsx`.

**Comment by LaChope** [comment]
Reopening as 1) is not complete.

**Comment by LaChope** [comment]
@blcham @kostobog 
We agreed that when clicking on the "layout" button, all persistence of nodes should be lost and the tree completely rearranged, the default behavior is always "manual" on.

**Comment by blcham** [comment]
@LaChope 
I understood it in a way that there is no `on` and `off` for layout. We always have a manual layout, but there is a button that automatically rearranges nodes only when clicked.

Hope we are saying same.

**Comment by LaChope** [comment]
Yes we do, there will not be any on/off, just one button to click to automatically layout the nodes.
